packages/gestalt/src/Dropdown.js

```diff
@@ -149,8 +160,13 @@ export default function Dropdown({
   onDismiss,
   zIndex,
   maxHeight,
+  mobileOnAnimationEnd,
+  mobileAdaptation = 'default',
 }: Props): Node {
-  const [hoveredItem, setHoveredItem] = useState<number>(0);
+  const deviceType = useDeviceType();
+  const isMobile = deviceType === 'mobile';
+
+  const [hoveredItem, setHoveredItem] = useState<?number>(isMobile ? undefined : 0);
```

Comment:
nit: this is the index of the hovered item, yes? If so, consider `hoveredItemIndex` to be more specific. We could also use an id or the item reference itself, so clarifying helps readability.

Reply:
DONE